selection.cpp

```diff
@@ -0,0 +1,210 @@
+    m_size = quiz.getCollectionOfQuestions().size();
+    m_answer.assign(m_size, 0);
+
+    emit Selection::play();
+}
+
+void Selection::playing()
+{
+    Selection::ButtonVisibility();
+    ui->comboBox_answers->clear();
+    ui->label_questionNumber->setText("Pitanje " + QString::number(indexOfCurrentQuestion + 1) + ":");
+    ui->label_question->setText(quiz.getCollectionOfQuestions()[indexOfCurrentQuestion].getQuestion());
+    for (int j = 0; j < 4; j++)
+    {
+        ui->comboBox_answers->addItem(quiz.getCollectionOfQuestions()[indexOfCurrentQuestion].getAnswer(j));
+    }
+    ui->comboBox_answers->setCurrentIndex(m_answer[indexOfCurrentQuestion] - 1);
+}
+
+void Selection::print()
+{
+    for(int k = 0; k < m_size; k++)
+    {
+        if(m_answer[k] == quiz.getCollectionOfQuestions()[k].getCorrectAnswer())
+        {
+            m_correctAnswers++;
+        }
+    }
+
+    ui->label_question->setText("Takmicar " + m_playerName + " je pogodio " + QString::number(m_correctAnswers) + "/"
+                                + QString::number(m_size) + " pitanja.");
+    ui->pushButton_highscore->setVisible(true);
+    ui->label_questionNumber->setVisible(false);
+    ui->comboBox_answers->setVisible(false);
+    ui->pushButton_next->setVisible(false);
+    ui->pushButton_previous->setVisible(false);
+    ui->pushButton_finish->setVisible(false);
+    ui->label_picture2->setVisible(false);
+    QPixmap picture3("://database/purple-confetti");
+    w = ui->label_confetti->width();
+    h = ui->label_confetti->height();
+    ui->label_confetti->setPixmap(picture3.scaled(w, h, Qt::IgnoreAspectRatio));
+
+    {
+        QSqlDatabase m_db = QSqlDatabase::addDatabase("QSQLITE");
+        m_db.setDatabaseName("/home/Sara/saraqt/qt_kviz/database/kviz.db");
+
+        if (!m_db.open())
+        {
+            qDebug() << "Error: connection with database failed";
+        }
+        else
+        {
+            qDebug() << "Database: connection ok";
+        }
+        QSqlQuery query4;
+        query4.prepare("INSERT INTO highscore VALUES ((:playerName), (:quizID), (:points))");
+        query4.bindValue(":playerName", m_playerName);
+        query4.bindValue(":quizID", quiz.getQuizID());
+        query4.bindValue(":points", m_correctAnswers);
+
+        query4.exec();
+
+        m_db.close();
+    }
+
+    QSqlDatabase::removeDatabase(QSqlDatabase::defaultConnection);
+
+}
+
+void Selection::on_pushButton_next_clicked()
+{
+    m_answer[indexOfCurrentQuestion] = ui->comboBox_answers->currentIndex() + 1;
+
+    indexOfCurrentQuestion++;
+
+    emit Selection::play();
+}
+
+
+void Selection::on_pushButton_previous_clicked()
+{
+    m_answer[indexOfCurrentQuestion] = ui->comboBox_answers->currentIndex() + 1;
+
+    indexOfCurrentQuestion--;
+
+    emit Selection::play();
+}
+
+void Selection::on_pushButton_finish_clicked()
+{
+    m_answer[indexOfCurrentQuestion] = ui->comboBox_answers->currentIndex() + 1;
+
+    emit Selection::finish();
+}
```

Comment:
Pošto je kod jako sličan u sve tri metode, istražiti da li je moguće uz pomoć nekog parametra objediniti ponašanje u jednu metodu.